src/features/surveys/components/SurveySubmissionsList.tsx

```diff
@@ -209,10 +203,13 @@ const SurveySubmissionsList = ({
       });
       setRespondentId(person?.id || null);
 
-      const personHasNoEmail = person?.id !== null && person?.email === null;
       const respondentEmail = row.respondent?.email;
-      if (personHasNoEmail && respondentEmail !== undefined) {
-        handleUpdateEmail(person, respondentEmail);
+      if (person) {
+        const personHasNoEmail = person.email == null || person.email == '';
+        if (personHasNoEmail && respondentEmail != undefined) {
```

Comment:
I think this is much nicer! I believe it could have been made even simpler, by relying on the fact that `null` and `undefined` are falsy, like so:

```suggestion
      if (person) {
        const personHasNoEmail = !person.email || person.email == '';
        if (personHasNoEmail && respondentEmail) {
```

But that's not worth changing the code over!